src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -316,11 +317,70 @@ class WelcomePage extends Disposable {
 		this.createPreviewModal();
 	}
 
+	private createButtons() {
+		const dropdownButtonContainer = document.querySelector('#dropdown_btn_container') as HTMLElement;
+		const nav = document.createElement('nav');
+		const dropdownUl = document.createElement('ul');
+		const i = document.createElement('div');
+
+		const newText = localize('welcomePage.new', "New");
+		let dropdownBtn = new Button(dropdownButtonContainer);
+		dropdownBtn.label = newText;
+
+		const iconClassList = ['twisties', 'codicon', 'codicon-chevron-right'];
+
+		i.classList.add(...iconClassList);
+
+		dropdownUl.classList.add('dropdown-content');
+		dropdownUl.setAttribute('aria-hidden', 'true');
+		dropdownUl.setAttribute('aria-label', 'submenu');
+		dropdownUl.setAttribute('role', 'menu');
+		dropdownUl.setAttribute('aria-labelledby', 'dropdown_btn');
+		dropdownUl.id = 'dropdown';
+		dropdownUl.innerHTML =
+			`<li role="none"><a role="menuitem" tabIndex="-1" class="move" href="command:registeredServers.addConnection">${(localize('welcomePage.newConnection', "New connection"))} </a></li>
+			<li role="none"><a role="menuitem" tabIndex="-1" class="move" href="command:workbench.action.files.newUntitledFile">${(localize('welcomePage.newQuery', "New query"))}</a></li>
+			<li role="none"><a role="menuitem" tabIndex="-1" class="move" href="command:notebook.command.new">${(localize('welcomePage.newNotebook', "New notebook"))}</a></li>
+			<li role="none" id="dropdown_mac_only"><a role="menuitem" tabIndex="-1" class="move mac_only" href="command:workbench.action.files.openLocalFileFolder">${(localize('welcomePage.openFileMac', "Open file"))}</a></li>
+			<li role="none" id="dropdown_windows_linux_only"><a role="menuitem" tabIndex="-1" class="move windows_only linux_only" href="command:workbench.action.files.openFile">${(localize('welcomePage.openFileLinuxPC', "Open file"))}</a></li`;
+		const getDropdownBtn = document.querySelector('#dropdown_btn_container .monaco-button') as HTMLElement;
+		getDropdownBtn.id = 'dropdown_btn';
+		getDropdownBtn.setAttribute('role', 'navigation');
+		getDropdownBtn.setAttribute('aria-haspopup', 'true');
+		getDropdownBtn.setAttribute('aria-controls', 'dropdown');
+		nav.setAttribute('role', 'navigation');
+		nav.classList.add('dropdown_nav');
+		dropdownUl.classList.add('dropdown');
+		getDropdownBtn.id = 'dropdown_btn';
+		getDropdownBtn.appendChild(i);
+		nav.appendChild(dropdownUl);
+		dropdownButtonContainer.appendChild(nav);
+
+		const fileBtnContainer = document.querySelector('#open_file_btn_container') as HTMLElement;
+		const openFileText = localize('welcomePage.openFile', "Open file");
+		let openFileMac = new Button(fileBtnContainer);
+		let openFileWindows = new Button(fileBtnContainer);
+		openFileWindows.label = openFileText;
+		openFileMac.label = openFileText;
+		dropdownBtn.label = newText;
+
+		const getNewFileBtnWindows = document.querySelector('#open_file_btn_container .monaco-button:first-of-type') as HTMLAnchorElement;
```

Comment:
Is there a reason you wouldn't just make a single button and set the appropriate values based on the platform?

Reply:
Hey @Charles-Gagnon I was just basing this on how VS code has theirs. I'm not sure how to set the values based on platform. Could you please advise? Thank you.

Reply:
https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/common/platform.ts#L107

Just switch on whether it's mac or not and then fill in the appropriate values based on that.

Reply:
okay I think it's okay now